controllers/helmchart_controller.go

```diff
@@ -526,9 +527,34 @@ func (r *HelmChartReconciler) reconcileFromTarballArtifact(ctx context.Context,
 		return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
 	}
 
+	v, err := semver.NewVersion(helmChart.Metadata.Version)
+	if err != nil {
+		err = fmt.Errorf("semver error: %w", err)
+		return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
+	}
+
+	version := v.String()
+	if chart.Spec.ReconcileStrategy == sourcev1.ReconcileStrategyRevision {
+		// Isolate the commit SHA from GitRepository type artifacts by removing the branch/ prefix.
+		re, err := regexp.Compile(".*/")
+		if err != nil {
+			err = fmt.Errorf("regex compile error: %w", err)
+			return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
+		}
+
+		v, err := v.SetMetadata(re.ReplaceAllString(artifact.Revision, ""))
```

Comment:
```suggestion
		splitRev := strings.Split(artifact.Revision, "/")
		v, err := v.SetMetadata(splitRev[(splitRev)-1])
```

Same result, no regex, no chance of returning an error.